app/Http/Controllers/ResultsController.php

```diff
@@ -66,7 +66,8 @@ public function index(MismatchGetRequest $request, WikibaseAPIClient $wikidata):
                 'user' => $user,
                 'item_ids' => $requestedItemIds,
                 // Use wikidata to fetch labels for found entity ids
-                'labels' => $wikidata->getLabels($entityIds, $lang),
+                // array_filter removes empty values from an array if no callback argument is passed
+                'labels' => $wikidata->getLabels(array_filter($entityIds), $lang),
```

Comment:
Under what circumstances would there be an empty element in `$entityIds`?

Reply:
in the case of the wikidata_value being empty. otherwise `getLabels` calls the WikibaseAPIClient with an invalid id, and it gives an error. you can try it out.

Reply:
From reading the code more, it seems the only way that an empty value can end up in `$entityIds` is if it is in the array returned by `extractItemIds`. Which in turn means we may want to adjust the condition in `extractItemIds` to actually only return item IDs:

```php
    private function extractItemIds(LazyCollection $mismatches, array $datatypes): array
    {
        return array_keys($mismatches->reduce(
            function (array $ids, Mismatch $mismatch) use ($datatypes) {
                $ids[$mismatch->item_id] = null;
                if ($datatypes[$mismatch->property_id] === 'wikibase-item') {
                    $ids[$mismatch->wikidata_value] = null;
                }
                return $ids;
            },
            []
        ));
    }
```

Reply:
I agree, it's better.